runtime/lua/vim/filetype.lua

```diff
@@ -1418,6 +1589,7 @@ local pattern = {
   ['.*/boot/grub/grub%.conf'] = 'grub',
   ['.*/boot/grub/menu%.lst'] = 'grub',
   ['.*/etc/grub%.conf'] = 'grub',
+  [vim.env.VIMRUNTIME .. '/doc/.*%.txt'] = 'help',
```

Comment:
Not sure if this will work, this file is compiled at build type so it will use whatever value of `VIMRUNTIME` is set at compile-time, not runtime.

Reply:
Ah, I forgot about the compilation. That's an issue, of course; then it has to be something like 
```lua
['.*/doc/.*%.txt'] = function(path,bufnr)
  if vim.fn.fnamemodify(path, ':h') == vim.env.VIMRUNTIME .. '/doc' then
    return 'help'
  end
end
```
(don't quote me on the modifier, though)...

Reply:
I'm actually second guessing myself now. I don't think the compilation into bytecode would actually index `vim.env`. That shouldn't happen until the module itself is loaded by the Lua VM, which is when Neovim is started. So maybe this actually does work 🙈

Reply:
Let's try it like this, and fix in a follow up :)